cmd/facts.go

```diff
@@ -108,7 +108,7 @@ func gather(*cobra.Command, []string) {
 		},
 	}
 
-	value, err := g.Gather(factRequest)
+	value, err := g.Gather(cmd.Context(), factRequest)
```

Comment:
Maybe I bit out of context, but we don't have any proper canceling on this command.
Could we maybe add something like this (copied from the `start` command):
```
signals := make(chan os.Signal, 1)
signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)

ctx, ctxCancel := context.WithCancel(context.Background())
// or
// ctx, ctxCancel := context.WithCancel(cmd.Context())

go func(ctx context.Context) {
	quit := <-signals
	log.Infof("Caught %s signal!", quit)

	log.Info("Stopping fact gathering...")
	ctxCancel()
}(ctx)

...

value, err := g.Gather(ctx, factRequest)
```

Reply:
I see this is coming in the next PR. We can discuss there.

Reply:
Indeed. 

I purposely avoid introducing behavior changes in this PR, as it's already fat.